File: solutions/[222]完全二叉树的节点个数.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def countNodes(self, root: TreeNode) -> int:
        def exists(root, level, k):
            bits = 1 << (level-1)
            
        node = root
        level = 0
        while node:
            level += 1
            node = node.left
        
        low, high = 1 << level, (1 << (level+1)) - 1
        while low < high:
            mid = (low+high) / 2
            if 
    

File: solutions/[222]完全二叉树的节点个数.cpp
#include <cstdlib>
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
	int countNodes(TreeNode* root) {
		if (root == nullptr)
			return 0;

		int depth = 0;
		return dp(root);
	}
private:
	int dp(TreeNode* root) {
		if (root == nullptr)
			return 0;

		int left_height = count_height(root->left);
		int right_height = count_height(root->right);
		if (left_height == right_height) {
			return (1 << left_height) + dp(root->right);
		}	
		else {
			return dp(root->left) + (1 << right_height);
		}
	}
	int count_height(TreeNode* root) {
		int height = 0;
		while (root != nullptr) {
			++height;
			root = root->left;
		}
		return height;
	}
};